src/components/pages/Receive.vue

```diff
@@ -62,43 +96,113 @@ export default {
       address: state =>
         state.accounts.address &&
         state.accounts.address.getChecksumAddressString(),
+      wallets: state => state.accounts.wallets,
+      activeNet: state => state.web3.activeNet,
     }),
     ...mapGetters('accounts', {
       balance: 'balance',
     }),
-    processedTransactions() {
-      const trxArr = [...this.transactions];
-
-      return trxArr.sort((trx1, trx2) => trx2.timestamp - trx1.timestamp);
+    ...mapGetters('transactions', ['currentNetTransactions']),
+    incomingTransactions() {
+      return (this.currentNetTransactions || []).filter(trx => {
+        return trx.to === this.address;
+      });
+    },
+    sortedTransactions() {
+      return this.incomingTransactions.sort(
+        (trx1, trx2) => trx2.timestamp - trx1.timestamp,
+      );
+    },
+  },
+  methods: {
+    ...mapActions('transactions', ['updateTransactionHistory']),
+    ...mapActions('accounts', ['selectWallet']),
+    async clickSendButton(address) {
+      this.selectWallet(address);
+      this.$router.push('/send');
+    },
+    isPublicWallet(wallet) {
+      return wallet instanceof Address;
+    },
+    isTokensLoaded(address) {
+      return typeof this.tokens[address] !== 'undefined';
+    },
+    async getHistory() {
+      if (!this.address) {
+        this.isLoading = false;
+        return;
+      }
+      this.isLoading = true;
+      await this.updateTransactionHistory();
+      this.isLoading = false;
+    },
+    getBalances(address) {
+      let addresses = Object.keys(this.wallets);
+      addresses.forEach(this.getBalance);
+    },
+    async getBalance(address) {
+      let balance = await this.wallets[address].getBalance();
+      balance = web3.utils.fromWei(balance);
+      this.$set(this.balances, address, balance);
+    },
+    getTokensLists(address) {
+      let addresses = Object.keys(this.wallets);
+      addresses.forEach(this.getTokensList);
```

Comment:
[The same story](https://github.com/endpass/endpass-wallet/pull/232/files#diff-7636d846f77116bfbee689b99fea7943R140)